app/views/contexts/summary_context.py

```diff
@@ -4,48 +4,112 @@
-            section_summary_context = SectionSummaryContext(
-                language=self._language,
-                schema=self._schema,
-                answer_store=self._answer_store,
-                list_store=self._list_store,
-                progress_store=self._progress_store,
-                metadata=self._metadata,
-                response_metadata=self._response_metadata,
-                current_location=location,
-                routing_path=self._router.routing_path(section_id),
-            )
-            section: Mapping = self._schema.get_section(section_id) or {}
-            if section.get("summary", {}).get("items"):
-                break
-
-            for group in section_summary_context(return_to=return_to)["summary"][
-                "groups"
-            ]:
-                yield group
+            if repeat := self._schema.get_repeat_for_section(section_id):
+                for_repeat = self._list_store[repeat["for_list"]]
+
+                if for_repeat.count > 0:
+                    for item in for_repeat.items:
+                        yield from self.build_summary_item(
+                            section_id=section_id,
+                            return_to=return_to,
+                            list_item_id=item,
+                            list_name=for_repeat.name,
+                        )
+            else:
+                yield from self.build_summary_item(
+                    section_id=section_id, return_to=return_to
+                )
+
+    def build_summary_item(
+        self,
+        section_id: str,
+        return_to: str | None,
+        list_name: str | None = None,
+        list_item_id: str | None = None,
+    ) -> Generator[ImmutableDict[str, Any], None, None]:
+        location = Location(
+            section_id=section_id, list_name=list_name, list_item_id=list_item_id
+        )
+        section_summary_context = SectionSummaryContext(
+            language=self._language,
+            schema=self._schema,
+            answer_store=self._answer_store,
+            list_store=self._list_store,
+            progress_store=self._progress_store,
+            metadata=self._metadata,
+            response_metadata=self._response_metadata,
+            current_location=location,
+            routing_path=self._router.routing_path(section_id),
+        )
+
+        return section_summary_context.build_summary(
+            return_to=return_to,
+            get_refactored_groups=False,
+            summary_type="Summary",
+            view_submitted_response=self.view_submitted_response,
+        ).get("groups", [])
+
+
+def set_unique_group_ids(groups: list[ImmutableDict]) -> list[ImmutableDict]:
+    checked_ids = set()
+    id_value = 0
+
+    for group in groups:
+        group_id = group["id"]
+        if group_id in checked_ids:
+            id_value += 1
+        checked_ids.add(group_id)
+        group["id"] = f"{group_id}-{id_value}"
+
+    return groups
```

Comment:
I needed to add something so the group ids were unique, adding a number rather than a list-item-id followed the same pattern we had in the section summary context and made it easy for the functional tests, as they use the same methods as section summary.

Reply:
Right okay makes sense, but given we have a method that changes the group ids, I suggest that is the single place we set group ids. It's confusing to me that we "refactor" the group and set unique group id (in the context of a single section) then we refactor again to set unique id (in the context of an entire survey). If we don't want to set the list item id to make testing easier, you could set the index of the list item id?. You have both the list store and location when calling `_get_refactored_groups`. This would mean that single methods handles refactors of groups including unique ids.